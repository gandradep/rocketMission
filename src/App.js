import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar';
import MyProfile from './components/MyProfile';
import RocketPage from './components/RocketsPage';
import MissionsPage from './components/MissionsPage';

function App() {
  return (
    <BrowserRouter>
      <div className="App">
        <header className="dFlex">
          <Navbar />
        </header>
        <section>
          <Routes>
            <Route path="/" element={<RocketPage />} />
            <Route path="/missions" element={<MissionsPage />} />
            <Route path="/myprofile" element={<MyProfile />} />
          </Routes>
        </section>
      </div>
    </BrowserRouter>
  );
}

export default App;
