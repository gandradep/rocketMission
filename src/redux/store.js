import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missions';
import rocketReducer from './rocketAction';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionsReducer,
  },
});

export default store;
