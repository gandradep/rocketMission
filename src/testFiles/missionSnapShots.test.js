import React from 'react';
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import  store from '../redux/store';
import App from '../App';

describe('Appcomponents snapshots', () => {
  it('App renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
