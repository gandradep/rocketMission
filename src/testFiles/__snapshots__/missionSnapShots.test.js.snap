// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appcomponents snapshots App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="dFlex"
        >
          <nav
            class="dFlex justify_between w-full items-center nav"
          >
            <div>
              <h1
                class="dFlex items-center"
              >
                <img
                  alt="Logo"
                  class="logo-img"
                  src="planet.png"
                />
                Space Traveler's Hub
              </h1>
            </div>
            <ul
              class="menuLinks dFlex"
            >
              <li>
                <a
                  aria-current="page"
                  class="activeLink"
                  href="/"
                >
                  Rockets
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/myprofile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </nav>
        </header>
        <section
          class="padSection"
        >
          <ul
            class="list"
          >
            <li />
          </ul>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="dFlex"
      >
        <nav
          class="dFlex justify_between w-full items-center nav"
        >
          <div>
            <h1
              class="dFlex items-center"
            >
              <img
                alt="Logo"
                class="logo-img"
                src="planet.png"
              />
              Space Traveler's Hub
            </h1>
          </div>
          <ul
            class="menuLinks dFlex"
          >
            <li>
              <a
                aria-current="page"
                class="activeLink"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class=""
                href="/myprofile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <section
        class="padSection"
      >
        <ul
          class="list"
        >
          <li />
        </ul>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
